docs/user/filesystem.md

```diff
@@ -894,3 +894,126 @@ $
 
 While this mode is not expected to be used directly by users, it is useful for
 the unified namespace integration.
+
+## Interception Library `libpil4dfs`
+
+A new interception library called `libpil4dfs` is introduced. It is similar to libioil,
+but it intercepts more APIs, including read/write as well as metadata related functions.
+This provides kernel-bypass for I/O data, leading to improved performance. This is a preview
+version. Some features are not implemented yet.
+
+### Using libpil4dfs with dfuse
+
+Start dfuse daemon,
+```
+dfuse -m /scratch_fs/dfuse tank mycont
+```
+
+To use the interception library, set `LD_PRELOAD` to point to the shared library
+in the DAOS install directory:
+
+```
+LD_PRELOAD=/path/to/daos/install/lib/libpil4dfs.so
+or
+LD_PRELOAD=/usr/lib64/libpil4dfs.so # when installed from RPMs
+```
+
+Example:
+
+```
+$ LD_PRELOAD=/usr/lib64/libpil4dfs.so mdtest -a POSIX -z 0 -F -C -i 1 -n 1667 -e 4096 -d /scratch_fs/dfuse/ -w 4096
+```
+### Print an interception summary
+
+If the `D_IL_REPORT` environment variable is set then the interception library will
+print a short summary of intercepted functions accessing DAOS filesystem through
+POSIX as a program exits. Both "D_IL_REPORT=1" and "D_IL_REPORT=true" enable printing
+the summary.
+
+```
+$ D_IL_REPORT=1 LD_PRELOAD=/usr/lib64/libpil4dfs.so mdtest -a POSIX -z 0 -F -C -i 1 -n 1667 -e 4096 -d /scratch_fs/dfuse -w 4096
+mdtest-3.4.0+dev was launched with 1 total task(s) on 1 node(s)
+Command line used: mdtest '-a' 'POSIX' '-z' '0' '-F' '-C' '-i' '1' '-n' '1667' '-e' '4096' '-d' '/scratch_fs/dfuse/' '-w' '4096'
+Path                : /scratch_fs/dfuse
+FS                  : 59.6 GiB   Used FS: 27.4%   Inodes: -0.0 Mi   Used Inodes: 0.0%
+Nodemap: 1
+1 tasks, 1667 files
+
+SUMMARY rate: (of 1 iterations)
+   Operation                     Max            Min           Mean        Std Dev
+   ---------                     ---            ---           ----        -------
+   File creation                2442.835       2442.835       2442.835          0.000
+   File stat                       0.000          0.000          0.000          0.000
+   File read                       0.000          0.000          0.000          0.000
+   File removal                    0.000          0.000          0.000          0.000
+   Tree creation                2957.901       2957.901       2957.901          0.000
+   Tree removal                    0.000          0.000          0.000          0.000
+-- finished at 06/20/2023 18:57:55 --
+
+libpil4dfs intercepting summary for ops on DFS:
+[read   ]  0
+[write  ]  1667
+
+[open   ]  1667
+[stat   ]  0
+[opendir]  0
+[readdir]  0
+[unlink ]  0
+[seek   ]  1667
+[mkdir  ]  2
+[rmdir  ]  0
+[rename ]  0
+[mmap   ]  0
+
+[op_sum ]  5003
+```
+
+### Using libpil4dfs with env
```

Comment:
please remove this section, or just mention this is for testing purposes only and not well tested/supported.

Reply:
This section is removed.